OrbitCore/PathTest.cpp

```diff
@@ -96,3 +96,171 @@ TEST(Path, JoinPathPartsAbsoluteRootBackslash) {
 #endif
   EXPECT_EQ(actual, expected);
 }
+
+TEST(Path, AutoInitBasePath) { EXPECT_NE(Path::GetBasePath(), ""); }
+
+TEST(Path, FileNamesAndExtensions) {
+#ifdef _WIN32
+  std::string input = "C:\\some_abs_path\\file.txt";
+  std::string stripped = "C:\\some_abs_path\\file";
+#else
+  std::string input = "/some_abs_path/file.txt";
+  std::string stripped = "/some/abs/path/file";
+#endif
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), stripped);
+  EXPECT_EQ(Path::GetFileName(input), "file.txt");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "file");
+
+  input = "relative/path/file.txt";
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), "relative/path/file");
+  EXPECT_EQ(Path::GetFileName(input), "file.txt");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "file");
+
+  input = "simple.file";
+  EXPECT_EQ(Path::GetExtension(input), ".file");
+  EXPECT_EQ(Path::StripExtension(input), "simple");
+  EXPECT_EQ(Path::GetFileName(input), "simple.file");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "simple");
+
+  input = "no/ext";
+  EXPECT_EQ(Path::GetExtension(input), "");
+  EXPECT_EQ(Path::StripExtension(input), "no/ext");
+  EXPECT_EQ(Path::GetFileName(input), "ext");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "ext");
+
+  input = "double.ext.txt";
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), "double.ext");
+  EXPECT_EQ(Path::StripExtension(Path::StripExtension(input)), "double");
+  EXPECT_EQ(Path::GetFileName(input), "double.ext.txt");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "double.ext");
```

Comment:
add test for StripExtenstion("/some/dir.with.dots/filename_without_dots");

Reply:
Good point, and that actually uncovered a bug. Added the test and fixed this issue.